lib/delete-users-using-local-file.rb

```diff
@@ -0,0 +1,135 @@
+require 'zendesk_api'
+require 'rest-client'
+require 'json'
+require 'date'
+
+@client = ZendeskAPI::Client.new do |config|
+  config.url = ENV['ZENDESK_URL']
+  config.username = ENV['ZENDESK_USER_EMAIL']
+  config.password = ENV['ZENDESK_USER_PASSWORD']
+  config.retry = true
+end
+
+# Setup vars
+
+# diag = "false"
+diag = "true"
+url = "#{ENV['ZENDESK_URL']}/deleted_users/"
+day = Date.today.next_day
+lastyear = day - 365
+false_date = "2012-01-01"
+
+input_file = File.open "data/soft_deleted_users.json"
+
+# This works for single line test file
+# file = File.open input_file
+# file_data = JSON.load file
+# file.close
+
+file_data = JSON.load input_file
+
+# file_data = JSON.parse(File.read(input_file).to_json)
+# puts file_data
+
+file_data.each do |user|
+
+
+  # Extract some fields
+  updated_at = user["updated_at"]
+  last_login_at = user["last_login_at"]
+  user_id = user["id"]
+  active = user["active"]
+  name = user["name"]
+
+  # Convert strings to dates, if strings are nil then set a false date of 2012 so comparison works.
+
+  if active == "false"
+    # account is already soft deleted so force hard delete
+    hard_delete = "true"
+  end
+
+  if hard_delete != "true"
+    if updated_at != nil
+      # parse the date so we can do comparisons
+      updated = Date.parse(updated_at)
+    else
+      # account has not been updated so consider for deletion
+      updated_at = false_date
+      updated = Date.parse(updated_at)
+    end
+
+    if last_login_at != nil
```

Comment:
See above.

Reply:
@blairboy362 Daniel, I have considered your change and prefer to keep the current construct for reading simplicity. Please resolve the conversation if you are happy with this or convince me otherwise :-)